src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -592,7 +599,9 @@ public void PushOpacity(double opacity, Rect? bounds)
         {
             CheckLease();
 
-            if(_useOpacitySaveLayer)
+            var useOpacitySaveLayer = _useOpacitySaveLayer || RenderOptions.RequiresFullOpacityHandling == true;
+
+            if (useOpacitySaveLayer)
             {
                 if (bounds.HasValue)
```

Comment:
So in this branch we still need to use the opacity stack, but instead of `_currentOpacity *= opacity;` we need `_currentOpacity = 1`